### PR TITLE
transfer endpoint

### LINKED ISSUE #145: Create transfer endpoint
The endpoint will be structured like `PUT /transfer/{resourceId}/{group}/{target}`

At present, the only value for _target_ is "ils". In the future, you can imagine it having additional values like "discovery".

The controller will:
* Verify that the user is a member of the group. (Note that the resource does NOT need to be associated with the group; a user can request the transfer of any resource regardless of the resource's groups.)
* Submit an SQS message requesting the transfer.

The name of queue will be `{group}-{target}`.

The message body will be serialized JSON with the following structure:
```
{
  "resource": {
    "uri": "{resourceURI -- full URI, not just resourceId}"
  },
  "user": {
    "email": "{user email}"
  },
  "group": "{group}",
  "target": "{target}"
}
```

The message will have an message attribute named `timestamp` that will contain the timestamp that the message is created formatted as ISO 8601.

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

closes #145

## How was this change tested?

* unit tests
* manual testing on the local command line (authenticate as my sinopia dev user, start the server with sinopia dev AWS profile, `curl -X POST -i -H "Authorization: Bearer $(cat .cognitoToken)" http://localhost:3000/transfer/foo/stanford/ils`)

## Which documentation and/or configurations were updated?

n/a

### CODE REVIEW

File: __tests__/endpoints/transferPost.test.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    })
+  })
+
+  describe("user is not in the target group", () => {
+    it("returns a 401 Unauthorized", async () => {
+      aws.buildAndSendSqsMessage.mockResolvedValue()
+      const res = await request(app)
+        .post("/transfer/foo/harvard/ils")
+        .set("Authorization", `Bearer ${stanfordJwt}`)
+        .send('')
```
Comment (jmartin-sul) [inline __tests__/endpoints/transferPost.test.js:53]: i can't figure out why, but this test seems to be hanging on me.  something in my authorization callback chain going wrong maybe?  works fine for the success case above.

**Comment by jmartin-sul** [comment]
fwiw, it seems like email is not always available in the JWT.  some debug logging from some local testing with my dev user:

```
    user: {
      sub: '2586583c-44bd-4d3c-9a6a-0a12345bc6de',
      'cognito:groups': [Array],
      event_id: '1464a32f-a4fa-4a29-abdd-6ccb10845c0b',
      token_use: 'access',
      scope: 'aws.cognito.signin.user.admin',
      auth_time: 1632877205,
      iss: 'https://cognito-idp.us-west-2.amazonaws.com/us-west-2_CGd9Wq136',
      exp: 1632880805,
      iat: 1632877205,
      jti: 'd4f39f14-fea3-4fd6-926e-d4f4bc566009',
      client_id: '2u6s7pqkc1grq1qs464fsi82at',
      username: 'jmartin-sul'
    },
```

File: __tests__/endpoints/transferPost.test.js (hunk 1 above)
Comment (justinlittman) [reply __tests__/endpoints/transferPost.test.js:53]: I've looked at this extensively and cannot determine the problem. You're OK to comment out the test and add a comment.

**Comment by justinlittman** [comment]
@jmartin-sul I don't see a `jmartin-sul` user in dev cognito. What email are you using to get token?

**Comment by jmartin-sul** [comment]
fixed the SQS message to include email

**Pull request merged**